### PR TITLE
Configurable nginx proxy prefix

### PR DESCRIPTION
**Pull request by mvdbeek**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -78,6 +78,8 @@ startup_chown_on_directory: ""
 nginx_conf_path: /etc/nginx/nginx.conf
 nginx_conf_directory: /etc/nginx/conf.d
 nginx_upload_store_path: "/tmp/nginx_upload_store"
+# Set nginx_proxy_prefix to "/galaxy", if galaxy should be served at subdirectory (e.g. example.com/galaxy)
+nginx_proxy_prefix: ""
```
Comment (jmchilton) [inline defaults/main.yml]: Can we take a finer pass at this - I'm thinking the default options file should look like:

```
nginx_prefix_location: ""
nginx_galaxy_location: "{{ nginx_prefix_location }}"
nginx_reports_location: "{{ nginx_prefix_location }}/reports"
nginx_planemo_web_location: "{{ nginx_prefix_location }}/planemo"
nginx_ide_location: "{{ nginx_prefix_location }}/ide"
```

And the rest of the PR should be updated accordingly. Then if you wanted to serve Galaxy on a prefix you'd probably just set:

```
nginx_galaxy_location: /galaxy
```

No reason to further prefix all the other apps. Reports etc... would still be "prefixed" at ``/reports``, ``/ide``, and ``/planemo``.

Comment (mvdbeek) [inline defaults/main.yml]: good idea. sure, will do.

Comment (jmchilton) [inline defaults/main.yml]: Awesome, thanks so much. This was on my TODO list (https://github.com/galaxyproject/planemo-machine/issues/53) so I'm looking forward to this - thanks for doing my job for me :smile:.

Comment (mvdbeek) [inline defaults/main.yml]: Great minds think alike :laughing: 
You'll still have to let the apps know that they run under a prefix though ;).

**Comment by mvdbeek** [comment]
Okay, I tested this also with @bgruening's image, if you run the playbook with --extra-vars "nginx_galaxy_location=/galaxy" and you change the galaxy.ini accordingly you get galaxy served at /galaxy. If you don't, nothing has changed.

**Comment by bgruening** [comment]
Awesome! Thanks!

**Comment by mvdbeek** [comment]
Thanks for merging @bgruening.

**Pull request merged**